Automattic/wp-calypso
# Staging Sites: add delete button on hosting config card

- Closes https://github.com/Automattic/dotcom-forge/issues/1757

## Proposed Changes

* Add delete button with a confirmation modal

## Testing Instructions



* Go to `/hosting-config/site-slug` 
* Click on `Create a staging site`
* Wait until the site is created
* Observe a new `Delete staging site` button
* Click on it
* Observe a new modal opens to confirm the operation
* Click on `Delete staging site` button inside the modal
* Observe the loading bar while the site is reverting
* Observe the `Create a staging site` appears after a few moments.

### Screencast





https://user-images.githubusercontent.com/779993/227293591-fc973a0f-6142-4f9f-8c1b-42280ef5669c.mp4





## Pre-merge Checklist
